### PR TITLE
feat: add sidebar component

### PR DESCRIPTION
**Pull request by chrisolsen**
DDI-1369

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
**Feedback from Tom**
- [ ] the hover state should be removed from the active page. (and only show active state) When it is on that page, the hover state for that page item in the menu should not be able to trigger, and it should stay as an active state. (there is no active hover state)
- [ ]  we need a bit of a different visual treatment. I have a version here that shows what is intended. https://www.figma.com/file/k9ZHctiEnN74GEhGWpio3K/Component---Side-Menu?type=design&node-id=2635-9810&t=svgdOOuEfmIMvxtb-4
Essentially every sub item is treated the same, with or without children (expand arrow), its indented another 2rem

**Review by vanessatran-ddi — DISMISSED** [review]
👍

**Review by vanessatran-ddi — APPROVED** [review]
👍

File: libs/styles/src/lib/reset.css
Comment (ArakTaiRoth) [inline libs/styles/src/lib/reset.css:1]: Do we need to make changes to this file? We don't use the styles lib anymore.

File: libs/docs/src/components/common/Sidebar.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+### The side menu is a navigation component positioned on the left side of the screen.
+
+[Figma design component](https://www.figma.com/file/k9ZHctiEnN74GEhGWpio3K/Component---Side-Menu?type=design&node-id=2541-9368&t=lnUt5z25Ypo1Fq5T-4)
+
+---
+
+#### SidebarGroup Properties
+
+<Props showTabs={true}>
+  <Prop name="label" type="string" description="Group header text" />
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/docs/src/components/common/Sidebar.stories.mdx:25]: I think we decided to always show the tabs, No?

[the remark this answers is not in the document]
Comment (ArakTaiRoth) [reply libs/docs/src/components/common/Sidebar.stories.mdx:25]: @chrisolsen Don't remember, but sounds good, doesn't really matter to me. The only important part is the prop name.

**Review by vanessatran-ddi — APPROVED** [review]
3rd 👍 :)

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**